components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -158,6 +187,8 @@ class AtmosphereInput
 
   std::shared_ptr<const FieldManager<Real>>   m_field_mgr;
   std::shared_ptr<const GridsManager>         m_grid_mgr;
+  grid_ptr_type                               m_grid;
+  bool                                        m_grid_set = false;
```

Comment:
Same comment as for the output  manager class: the bool is superfluous, and can be replaced with a direct check on the ptr.